airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -404,28 +427,81 @@ async def _build_regression_test_container(self, target_container_id: str) -> Co
-                .with_mounted_file("/root/.ssh/id_rsa", self.dagger_client.host().file(str(Path("~/.ssh/id_rsa").expanduser())))  # TODO
+            container.with_exec(["apt-get", "update"])
+            .with_exec(container_requirements)
+            .with_exec(["bash", "-c", "curl https://sdk.cloud.google.com | bash"])
+            .with_env_variable("PATH", "/root/google-cloud-sdk/bin:$PATH", expand=True)
+            .with_mounted_directory("/app", self.context.live_tests_dir)
+            .with_workdir("/app")
+            .with_unix_socket("/var/run/docker.sock", self.dagger_client.host().unix_socket("/var/run/docker.sock"))
+            .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # The connector being tested is already built and is stored in a location accessible to an inner dagger kicked off by
+            # regression tests. The connector can be found if you know the connection ID, so we write the connection ID to a file and put
+            # it in the regression test container. This way regression tests will use the already-built connector instead of trying to
+            # build their own.
+            .with_new_file("/tmp/container_id.txt", contents=str(target_container_id))
+            .with_exec(["pip", "install", "poetry"])
+        )
+
+        if self.context.is_ci:
+            container = (
+                container
+                # In CI, use https to get the connection-retriever package from airbyte-platform-internal instead of ssh
+                .with_exec(
+                    [
+                        "sed",
+                        "-i",
+                        "-E",
+                        r"s,git@github\.com:airbytehq/airbyte-platform-internal,https://github.com/airbytehq/airbyte-platform-internal.git,",
+                        "pyproject.toml",
+                    ]
+                )
+                .with_exec(
+                    ["poetry", "source", "add", "--priority=supplemental", "airbyte-platform-internal-source",
+                     "https://github.com/airbytehq/airbyte-platform-internal.git"]
+                ).with_exec(
+                    ["poetry", "config", "http-basic.airbyte-platform-internal-source", "octavia-squidington-iii",
+                     self.context.ci_github_access_token]
+                )
+                # Add GCP credentials from the environment and point google to their location (also required for connection-retriever)
+                .with_new_file("/tmp/credentials.json", contents=os.getenv("GCP_INTEGRATION_TESTER_CREDENTIALS"))
+                .with_env_variable("GOOGLE_APPLICATION_CREDENTIALS", "/tmp/credentials.json")
+                # Set up Google Cloud SQL Proxy to connect to the database
+                .with_exec([
+                    "curl",
+                    "-o",
+                    "cloud-sql-proxy",
+                    "https://storage.googleapis.com/cloud-sql-connectors/cloud-sql-proxy/v2.11.0/cloud-sql-proxy.linux.amd64",
```

Comment:
Nit: as this is platform dependent you can dynmacially build  the URL according[ to the current platform](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/consts.py#L31)